include/extractor/extraction_way.hpp

```diff
@@ -50,6 +65,20 @@ struct ExtractionWay
     void set_backward_mode(const TravelMode m) { backward_travel_mode = m; }
     TravelMode get_backward_mode() const { return backward_travel_mode; }
 
+    // wrappers to allow assigning nil (nullptr) to string values
+    void SetName(const char* value) { detail::maybeSetString(name, value); }
```

Comment:
Hrm we need a get/set function for every member. The business logic is already moved to a helper function. I don't see how we could optimize this further.